test/include.test.js

```diff
@@ -190,19 +190,322 @@ describe('include', function () {
+        User.findOne({
+          include: {
+            relation: 'passports'
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          user.passports()[0].id.should.equal(1);
+          user.passports()[0].ownerId.should.equal(1);
+          user.passports()[0].number.should.equal('1');
+
+          done();
+        });
+      });
+
+      it('works when using a `where` with `and`', function(done) {
+        User.findOne({
+          include: {
+            relation: 'posts',
+            scope: {
+              where: {
+                and: [
+                  {id: 1},
+                  {userId: 1},
+                  {title: 'Post A'}
+                ]
+              }
+            }
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          user.name.should.equal('User A');
+          user.age.should.equal(21);
+          user.id.should.equal(1);
+          posts = user.posts();
+          posts.length.should.equal(1);
+          posts[0].title.should.equal('Post A');
+          posts[0].userId.should.equal(1);
+          posts[0].id.should.equal(1);
+
+          done();
+        });
+      });
+
+      it('works when using `where` with `limit`', function(done) {
+        User.findOne({
+          include: {
+            relation: 'posts',
+            scope: {
+              limit: 1
+            }
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          user.posts().length.should.equal(1);
+
+          done();
+        });
+      });
+
+      it('works when using `where` with `skip`', function(done) {
+        User.findOne({
+          include: {
+            relation: 'posts',
+            scope: {
+              skip: 1
+            }
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          var posts = user.posts();
+          posts.length.should.equal(2);
+          posts[0].id.should.equal(2);
+          posts[1].id.should.equal(3);
+
+          done();
+        });
+      });
+
+      it('works when using `where` with `offset`', function(done) {
+        User.findOne({
+          include: {
+            relation: 'posts',
+            scope: {
+              offset: 1
+            }
+          }
+        }, function(err, user) {
+          if (err) return done(err);
+
+          var posts = user.posts();
+          posts.length.should.equal(2);
+          posts[0].id.should.equal(2);
+          posts[1].id.should.equal(3);
+
+          done();
+        });
+      });
+
+      it('works when using `where` without `limit`, `skip` or `offset`',
+          function(done) {
+        User.findOne({include: {relation: 'posts'}}, function(err, user) {
+          if (err) return done(err);
+
+          var posts = user.posts();
+          posts[0].id.should.equal(1);
+          posts[1].id.should.equal(2);
+          posts[2].id.should.equal(3);
```

Comment:
These three asserts can be simplified to the following, it will also give you more helpful message when e.g. the ids are sorted differently:

```js
var ids = posts.map(function(p) { return p.id; }); // or use _.property('id') from lodash
ids.should.eql([1,2,3]);
```

The same comment applies to other tests below.